### ISSUE TITLE
Importing the entrypoint with splitting on produces broken bundle

### ISSUE THREAD

**Issue by paperdave** [evanw/esbuild#3043]
Given `entry.js`:
```ts
console.log(await import("./entry.js"));
```

I don't know why this behavior would ever be used, but it doesn't crash node or bun. esbuild produces this when bundled with `--bundle --format=esm --splitting --outdir=out`

```ts
// entry.js
console.log(await require_entry());
```

which crashes at runtime.

This should be tested in `TestTopLevelAwaitAllowedImportWithSplitting` but the snapshot also contains broken code.
snapshot: https://github.com/evanw/esbuild/blob/main/internal/bundler_tests/snapshots/snapshots_default.txt#L5635
test source: https://github.com/evanw/esbuild/blob/main/internal/bundler_tests/bundler_default_test.go#L4149

**Comment by hyrious** [comment]
> but it doesn't crash node or bun

Nope, I just tried that and it seems Node.js would exit with non-zero code on circular imports.

```js
// a.mjs
console.log(1);
await import("./a.mjs");
console.log(2);
```

```console
$ node a.mjs
1
$ echo $?
13
```

But this case is still possible to be bundled in a bundler. Rollup generates a promise that returns an uninitialized var (== undefined): [repl link](https://rollupjs.org/repl/?shareable=JTdCJTIyZXhhbXBsZSUyMiUzQSUyMiUyMiUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJjb2RlJTIyJTNBJTIyYXdhaXQlMjBpbXBvcnQoJy4lMkZiLmpzJyklMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSUyQyUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTdEJTJDJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU1QyUyMi4lMkZtYWluLmpzJTVDJTIyJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTJDJTIybmFtZSUyMiUzQSUyMmIuanMlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyb3V0cHV0JTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlMkMlMjJpbmxpbmVEeW5hbWljSW1wb3J0cyUyMiUzQXRydWUlN0QlN0QlN0Q=)

**Comment by paperdave** [comment]
you're right with node actually erroring on it, didn't look close enough at the status code, just saw no text being printed. but yes it should still be bundlable, or it should cause a build time error.

**Comment by hyrious** [comment]
I just came up with a runnable example that should be handled:

```js
// a.mjs
export let a = import("./a.mjs").then(console.log);
```

This will print as below in Node.js:

```js
[Module: null prototype] { a: Promise { <pending> } }
```

..but not when bundled with `--splitting` &darr; [repl link](https://hyrious.me/esbuild-repl/?version=0.17.15&mode=build&modules=%5B%5B%22a.mjs%22%2C%22export+let+a+%3D+import%28%5C%22.%2Fa.mjs%5C%22%29.then%28console.log%29%22%2C1%5D%5D)

```js
// a.mjs
var a = Promise.resolve().then(() => require_a()).then(console.log);
export {
  a
};
```

Without `--splitting` it works well.
